### ISSUE TITLE
Figure out a way for an exercise / set to have an arbitrary number of properties

### ISSUE THREAD

**Issue by GideonNeedleman** [GideonNeedleman/5x5#22]
Most likely can do this using a many : many junction table between exercises & properties. So you have a properties table with things like weight, reps, time, distance, height (for high jumps or something), difficulty (for settings on a machine or something), laps, etc. The junction table then holds FK for both the exercise & properties.

_exercises_
id
name
description
FK owner
tags

_properties_
id
name
units (will likely need a units table to allow conversions)

_exercise-property-junction-table_
id
FK exercise
FK property

This should allow you to specify arbitrary number of properties for any exercise, but how do individual sets fit into this schema? Each set needs to record a default value for each property as well as its own ID, a FK to exercise and timestamp.

_sets_
id
FK exercise
timestamp

_set-property-junction-table_
id
FK set
FK property
value (could be an integer quantity or float distance or a difficulty string)

Basically every time you add a set to an exercise, it needs to lookup all the properties of that exercise and create rows in the set-property-JT to replicate each property. Would it be possible to merge the exercise-property-JT and the set-property-JT? 

_exercise-set-property-JT_
id
FK exercise
FK set
FK property
value

In this case, if you're defining an exercise then the FK set field is null.
If you're listing a set, the FK exercise field is null. 

Or if you're listing a set there is no null field, but instead remove the FK exercise field from the sets table since it's now redundant.
